File: .claude/hooks/permission-auto-allow.mjs
#!/usr/bin/env node
// @hookstack permission-request-auto-allow-readonly
// Auto-autorise les outils lecture seule et les commandes Bash sûres (PermissionRequest)
import { readFileSync } from 'node:fs';
import { fileURLToPath } from 'node:url';

const SAFE_BASH = [
  /^ls/, /^pwd/, /^echo/, /^cat(?!.*>)/, /^head/, /^tail/, /^wc/,
  /^which/, /^whereis/, /^file/, /^stat/,
  /^git\s+(status|log|diff|show|branch|tag)/,
  /^npm\s+(list|ls|outdated|view)/,
];

const READ_ONLY_TOOLS = ['Read', 'Glob', 'Grep'];

export function run(input) {
  const toolName = input.tool_name || '';
  const toolInput = input.tool_input || {};

  let allow = READ_ONLY_TOOLS.includes(toolName);
  if (!allow && toolName === 'Bash') {
    const cmd = (toolInput.command || '').trim();
    allow = SAFE_BASH.some((p) => p.test(cmd));
  }

  return allow
    ? {
        hookSpecificOutput: {
          hookEventName: 'PermissionRequest',
          decision: { behavior: 'allow' },
        },
      }
    : null;
}

/* v8 ignore next 5 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  const result = run(input);
  if (result) process.stdout.write(JSON.stringify(result));
}

File: .claude/hooks/permission-auto-allow.py
#!/usr/bin/env python3
# @hookstack permission-request-auto-allow-readonly
"""Auto-allows read-only tools and safe Bash commands (PermissionRequest)."""
import json
import re
import sys

SAFE_BASH = [
    re.compile(r"^ls"),
    re.compile(r"^pwd"),
    re.compile(r"^echo"),
    re.compile(r"^cat(?!.*>)"),
    re.compile(r"^head"),
    re.compile(r"^tail"),
    re.compile(r"^wc"),
    re.compile(r"^which"),
    re.compile(r"^whereis"),
    re.compile(r"^file"),
    re.compile(r"^stat"),
    re.compile(r"^git\s+(status|log|diff|show|branch|tag)"),
    re.compile(r"^npm\s+(list|ls|outdated|view)"),
]

READ_ONLY_TOOLS = ["Read", "Glob", "Grep"]


def run(input_data=None):
    data = input_data or {}
    tool_name = data.get("tool_name") or ""
    tool_input = data.get("tool_input") or {}

    allow = tool_name in READ_ONLY_TOOLS
    if not allow and tool_name == "Bash":
        cmd = (tool_input.get("command") or "").strip()
        allow = any(p.search(cmd) for p in SAFE_BASH)

    if not allow:
        return None
    return {
        "hookSpecificOutput": {
            "hookEventName": "PermissionRequest",
            "decision": {"behavior": "allow"},
        },
    }


if __name__ == "__main__":
    data = json.load(sys.stdin)
    result = run(data)
    if result:
        sys.stdout.write(json.dumps(result))
